### PR TITLE
Introduce bounding box selection component

### PR DESCRIPTION
**Pull request by ahennr**
This PR covers the bounding box selection component as defined in #26.

image

Within the map users can now adjust the spatial extent of his query using a modifiable bounding box.
This box is activated by clicking on the map. It controls the filter `q.geo` whereas the heat map itself (parameter `a.hm.filter`) is always requested for the extent of the map that is currently visible.
If the query box becomes greater than the map extent (f.e. if user zooms in) it will be reduced to 90% of the current map extent (configurable in `appConfig.json` - parameter `ratioInnerBbox`) in order to keep it visible entirely.

All layers outside the bounding box are masked white.

The text field indicating the current Filter `q.geo` (name "Geo") will be addressed in a future PR.

Please review @terrestris/owners.

### CODE REVIEW

File: config/appConfig.json [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
         "csvDocsLimit": 1
     },
     "mapConfig": {
         "renderer": "canvas",
         "view": {
-            "center": [430493, 6887893],
-            "maxZoom": 10,
-            "minZoom": 2,
+            "center": [0, 0],
+            "extent": [-20000000.00,-20000000.00,20000000.00,20000000.00],
```
Comment (annarieger) [inline config/appConfig.json]: Where do you have this values from? Shouldn't we use the "official" mercator bounds extent here instead? (`[-20037508.34,-20037508.34,20037508.34,20037508.34]`)

**Comment by annarieger** [comment]
A great one, I've added only one little question :+1: 

Please merge

File: config/appConfig.json (hunk 1 above)
Comment (ahennr) [inline config/appConfig.json]: This extent is especially used for the initial extent of the query bounding box. I changed this to official mercator bounds in 5c46a22

**Comment by ahennr** [comment]
THX for review @annarieger . Will merge when all checks are green.

**Pull request merged**